packages/core-components/src/layout/HeaderLabel/HeaderLabel.tsx

```diff
@@ -49,12 +49,19 @@ const useStyles = makeStyles<BackstageTheme>(
 type HeaderLabelContentProps = PropsWithChildren<{
   value: React.ReactNode;
   className: string;
+  typographyRootComponent?: keyof JSX.IntrinsicElements;
 }>;
 
-const HeaderLabelContent = ({ value, className }: HeaderLabelContentProps) => {
+const HeaderLabelContent = ({
+  value,
+  className,
+  typographyRootComponent,
+}: HeaderLabelContentProps) => {
   return (
     <Typography
-      component={typeof value === 'string' ? 'p' : 'span'}
+      component={
+        typographyRootComponent ?? (typeof value === 'string' ? 'p' : 'span')
```

Comment:
Suggestion:

What do you think about also redefining typography styles regardless of the component used here? I mean, the appearance be the same regardless of the component used here.

The components that are rendering the label (page header, entity card, etc.) could provide a class name that modifies the spacing between elements if desired.

Reply:
That was also an option I considered. Within the scope of these few components, I did not see any other examples of passing custom classes from the parent, so I decided to use this method instead.
Is that what you would recommend?

With this implementation, the header item content are now both nested inside `p` tags.
So for example in the screenshot, the `a-team` link is inside a `p` (instead of a `span`). This matches with the `experimental` lifestyle to the right of it, which is also in `p` tag.

<img width="240" alt="image" src="https://github.com/backstage/backstage/assets/30599893/bb63d156-5b25-49df-85b0-d70304bbc770">

Reply:
I was thinking more about having a `className` in the root `Typography` component and we can use CSS Grid or Flexbox to modify the spacing between child components (label and content) without necessarily forwarding a class to them. But I haven't tested the idea, so it's just a suggestion. What do you think @backstage/maintainers?

Reply:
Yes definitely makes sense. Passing down a classname to the `Typography` component instead of the typography's `component` props works as well.